packages/react-router/lib/components.tsx

```diff
@@ -220,49 +223,28 @@ export function Outlet(props: OutletProps): React.ReactElement | null {
   return useOutlet(props.context);
 }
 
-interface DataRouteProps {
-  id?: RouteObject["id"];
-  loader?: RouteObject["loader"];
-  action?: RouteObject["action"];
-  errorElement?: RouteObject["errorElement"];
-  shouldRevalidate?: RouteObject["shouldRevalidate"];
-  handle?: RouteObject["handle"];
-}
-
-export interface RouteProps extends DataRouteProps {
-  caseSensitive?: boolean;
-  children?: React.ReactNode;
-  element?: React.ReactNode | null;
-  index?: boolean;
-  path?: string;
-}
+// Create JSX-specific types from the RouteObjects, essentially changing
+// ReactElement to ReactNode
+export type PathRouteProps = Omit<PathRouteObject, "children"> & {
+  children?: React.ReactNode | null;
+};
```

Comment:
`<Route>` props are now just the underlying `RouteObject` types with `ReactElement -> ReactNode`